src/components/structure/visualizing/artefact-overview/artefact-overview.tsx

```diff
@@ -159,14 +158,24 @@ const ArtefactOverview: FC<OverviewProps> & { Switcher: FC<SwitcherProps> } = ({
     const safetyDistance = 1;
     const rows = Math.floor(artefactOverviewDimensions.height / tileSize) - safetyDistance;
     const cols = Math.floor(artefactOverviewDimensions.width / tileSize);
+
     return cols * rows;
   }
 
+  const setFittedArtefactAmount = () => {
+    const element = artefactOverviewElRef.current;
+    const suitableAmountOfArtefacts = getSuitableAmountOfArtefacts(element);
+    handleArtefactAmountChange(suitableAmountOfArtefacts);
+  }
+
   useEffect(() => {
     if (!artefactOverviewElRef.current) return;
 
     if (viewType === ArtefactOverviewType.CARD_SMALL) {
-      handleArtefactAmountChange(getSuitableAmountOfArtefacts(artefactOverviewElRef.current));
+      setFittedArtefactAmount();
+      window.addEventListener('resize', function () {
```

Comment:
Hier müsste man aufpassen.

Wenn sich `viewType` und `artefactOverviewElRef` öfters ändern, wird auch der Effect ausgelöst.
Ändert sich hierbei dann auch noch der `viewType`-Value von `ArtefactOverviewType.CARD_SMALL` zu einem anderen Value und zurück, dann wird jedes Mal `window.addEventListener` aufgerufen, ohne den Listener vorher wieder zu entfernen, sofern zuvor schon ein Listener hinzugefügt wurde.

Im schlechtesten Falle, werden bei jedem Resize mehrere Listener-Funktionen ausgelöst.

Um das Problem zu umgehen, sollte die Listener-Funktion nicht mehr anonym sein (um sie referenzieren zu können) und nach dem `window.addEventListener` eine Funktion zurückgegeben werden, mit deren Aufruf der Listener vor dem erneuten Auslösen des Effects erstmal wieder entfernt wird.

Also in etwa so:

```typescript
useEffect(() => {
  if (!artefactOverviewElRef.current) return;

  if (viewType === ArtefactOverviewType.CARD_SMALL) {
    setFittedArtefactAmount();
    window.addEventListener('resize', setFittedArtefactAmount);

    return () => window.removeEventListener('resize', setFittedArtefactAmount);
  }
}, [viewType, artefactOverviewElRef]);
```